components/automate-ui/src/app/pages/+compliance/+reporting/+reporting-overview/reporting-overview.component.ts

```diff
@@ -114,6 +114,63 @@ export class ReportingOverviewComponent implements OnInit, OnDestroy {
     }
   }
 
+  onPlatformChanged(platformItem) {
+    const typeName = 'platform';
+    if (platformItem && platformItem.name) {
+      const {queryParamMap} = this.route.snapshot;
+      const queryParams = {...this.route.snapshot.queryParams};
+      const existingValues = queryParamMap.getAll(typeName).filter(
+        v => v !== platformItem.name).concat(platformItem.name);
+
+      queryParams[typeName] = existingValues;
+
+      this.router.navigate(['/compliance', 'reports', 'nodes'], {queryParams});
+    }
+  }
+
+  onEnvironmentChanged(environmentItem) {
+    const typeName = 'environment';
+    if (environmentItem && environmentItem.name) {
+      const {queryParamMap} = this.route.snapshot;
+      const queryParams = {...this.route.snapshot.queryParams};
+      const existingValues = queryParamMap.getAll(typeName).filter(
+        v => v !== environmentItem.name).concat(environmentItem.name);
+
+      queryParams[typeName] = existingValues;
+
+      this.router.navigate(['/compliance', 'reports', 'nodes'], {queryParams});
+    }
+  }
+
+  onProfileChanged(profileItem) {
+    const typeName = 'profile_id';
+    if (profileItem && profileItem.id) {
+      const {queryParamMap} = this.route.snapshot;
+      const queryParams = {...this.route.snapshot.queryParams};
+      const existingValues = queryParamMap.getAll(typeName).filter(
+        v => v !== profileItem.id).concat(profileItem.id);
+
+      this.reportQuery.setFilterTitle(typeName, profileItem.id, profileItem.name);
+      queryParams[typeName] = existingValues;
+
+      this.router.navigate(['/compliance', 'reports', 'nodes'], {queryParams});
+    }
+  }
+
+  onControlChanged(controlItem) {
+    const typeName = 'control_id';
+    if (controlItem && controlItem.name) {
+      const {queryParamMap} = this.route.snapshot;
+      const queryParams = {...this.route.snapshot.queryParams};
+      const existingValues = queryParamMap.getAll(typeName).filter(
+        v => v !== controlItem.name).concat(controlItem.name);
+
+      queryParams[typeName] = existingValues;
+
+      this.router.navigate(['/compliance', 'reports', 'nodes'], {queryParams});
+    }
+  }
```

Comment:
Since these 4 functions are all the same except for the type and field names, you could potentially extract the logic to a private function that these 4 called with the appropriate values. It'd make it clear at a glance that these are all doing the same thing, but I'm not sure if that'd be too abstract and would hurt readability though.

Reply:
Nice call! That cleaned out the code well.